path_roles.go

```diff
@@ -132,23 +141,33 @@ func (b *tfBackend) pathRolesWrite(ctx context.Context, req *logical.Request, d
 		roleEntry = &terraformRoleEntry{}
 	}
 
-	roleEntry.Name = name
+	createOperation := (req.Operation == logical.CreateOperation)
 
+	roleEntry.Name = name
 	if organization, ok := d.GetOk("organization"); ok {
 		roleEntry.Organization = organization.(string)
-	} else if req.Operation == logical.CreateOperation {
+	} else if createOperation {
 		roleEntry.Organization = d.Get("organization").(string)
-		if roleEntry.Organization == "" {
-			return logical.ErrorResponse("missing organization"), nil
-		}
 	}
 
-	if team, ok := d.GetOk("team_id"); ok {
-		roleEntry.TeamID = team.(string)
-	} else if team != nil {
+	if teamID, ok := d.GetOk("team_id"); ok {
+		roleEntry.TeamID = teamID.(string)
+	} else if createOperation {
 		roleEntry.TeamID = d.Get("team_id").(string)
-	} else {
-		roleEntry.TeamID = ""
+	}
+
+	if userID, ok := d.GetOk("user_id"); ok {
+		roleEntry.UserID = userID.(string)
+	} else if createOperation {
+		roleEntry.UserID = d.Get("user_id").(string)
+	}
+
+	if roleEntry.UserID != "" && (roleEntry.Organization != "" || roleEntry.TeamID != "") {
+		return logical.ErrorResponse("cannot provide a user_id in combination with organization or team_id"), nil
+	}
+
+	if roleEntry.UserID == "" && roleEntry.Organization == "" && roleEntry.TeamID == "" {
+		return logical.ErrorResponse("must provide an organization name, team id, or user id"), nil
```

Comment:
Should we be checking for the case in which a person provides a teamId, but no organization?

Reply:
`organization` isn't required to make a team role. It's technically _allowed_, but it's not required nor even used if a `team_id` is given. The TFC API for teams is not nested/related to an organization, so you don't need to know the organization a team belongs to in order to interact with the team via the API.